govstack_api/building_blocks/bb_digital_registries/registries/insuree_registry.py

```diff
@@ -1,117 +1,29 @@
-                      otherNames: "Tatra"
-                      {family_id}
-                      {gender_id}
-                      {head}
-                      {card_issued}
-                    }}
-                  ) {{
-                    clientMutationId
-                    internalId
-                  }}
-                }}
-                '''
-    return query
-
-
-#TODO make it that it will get required fields with some data and then compare if the data is present in mapped that, if no then use. If else the use those fields that are not present inside mapped data
-def get_create_registry_query(
-        arguments_with_values: str = "",
-        family_id="familyId: 1",
-        gender_id='genderId: "U"',
-        dob=config.date_of_birth,
-        head='head: false',
-        card_issued='cardIssued: false'
-) -> str:
-    return f'''
-        mutation {{
-          createInsuree(
-            input: {{
-              clientMutationId: "9d738f06-7855-455a-840f-99aa85d2308e"
-              clientMutationLabel: "Create insuree"
-              chfID: "{arguments_with_values['id']}"
-              {arguments_with_values}
-              {family_id}
-              {gender_id}
-              {dob}
-              {head}
-              {card_issued}
-            }}
-          ) {{
-            clientMutationId
-            internalId
-          }}
-        }}
-        '''
-
-
-class InsureeRegistry(BaseRegistry):
-
-    def __init__(self, request):
+class InsureeRegistry(BaseRegistry, RegistryProtocol):
+
+    def __init__(self, registry_config, request):
+        super().__init__(registry_config, request)
         self.client = GrapheneClient(request, Query, Mutation)
-        self.fields_mapping = {
-            'ID': 'id',
-            'FirstName': 'otherNames',
-            'LastName': 'lastName',
-            'uuid': 'uuid'
-        }
-        self.special_fields = ['BirthCertificateID', 'PersonalData']
 
     def get_record(self, mapped_data, fetched_fields=None, only_first=True):
         # workaround because for now we lack on filtering json_ext
         if not fetched_fields:
             fetched_fields = self.create_fetched_fields(mapped_data)
         mapped_data.pop('jsonExt', None)
-
-        mapped_data = self.encode_id(mapped_data)
-        arguments_with_values = ', '.join(f'{field}: "{value}"' for field, value in mapped_data.items())
-        query = get_insurees_query(arguments_with_values, fetched_fields)
+        mapped_data = self.insuree_encode_id(mapped_data)
+        arguments_with_values = self.create_arguments_with_values(mapped_data)
+        query = self.get_single_model_query(self.queries["get"], arguments_with_values, fetched_fields)
         result = self.client.execute_query(query)
-        registry_data = self.extract_records(result, only_first)
+        registry_data = self.extract_insuree_records(result, self.queries['get'], only_first)
         return registry_data
```

Comment:
I think that whole get_record function could be in base registry. 
self.insuree_encode_id could be replaced with `self.encode_id(mapped_data, [parameter_from_registry_config])`.  
`create_arguments_with_values` also looks like it could have generic implementation in the base class. 
`self.extract_insuree_records` is an unnecessary wrapper for `self.extract_records(result, query_get, only_first)`.